h3i/src/client/connection_summary.rs

```diff
@@ -213,41 +300,110 @@ impl StreamMap {
     /// assert!(!stream_map.is_empty());
     /// ```
     pub fn is_empty(&self) -> bool {
-        self.0.is_empty()
+        self.map.is_empty()
     }
 
-    /// See all HEADERS received on a given stream.
-    ///
-    /// # Example
-    ///
-    /// ```
-    /// use h3i::client::connection_summary::StreamMap;
-    /// use h3i::frame::EnrichedHeaders;
-    /// use h3i::frame::H3iFrame;
-    /// use quiche::h3::Header;
-    /// use std::iter::FromIterator;
-    ///
-    /// let mut stream_map = StreamMap::default();
-    ///
-    /// let h = Header::new(b"hello", b"world");
-    /// let enriched = EnrichedHeaders::from(vec![h]);
-    /// let headers = H3iFrame::Headers(enriched.clone());
-    /// let data = H3iFrame::QuicheH3(quiche::h3::frame::Frame::Data {
-    ///     payload: b"hello world".to_vec(),
-    /// });
-    ///
-    /// let stream_map: StreamMap = [(0, vec![headers.clone(), data.clone()])].into();
-    /// assert_eq!(stream_map.headers_on_stream(0), vec![enriched]);
-    /// ```
-    pub fn headers_on_stream(&self, stream_id: u64) -> Vec<EnrichedHeaders> {
-        self.stream(stream_id)
-            .into_iter()
-            .filter_map(|h3i_frame| h3i_frame.to_enriched_headers())
-            .collect()
+    pub(crate) fn new(expected_frames: Option<ExpectedFrames>) -> Self {
+        Self {
+            expected_frames,
+            ..Default::default()
+        }
     }
 
     pub(crate) fn insert(&mut self, stream_id: u64, frame: H3iFrame) {
-        self.0.entry(stream_id).or_default().push(frame);
+        if let Some(expected) = self.expected_frames.as_mut() {
+            expected.receive_frame(stream_id, &frame);
+        }
+
+        self.map.entry(stream_id).or_default().push(frame);
+    }
+
+    pub(crate) fn check_expected_frames(&self) -> bool {
```

Comment:
Lets document this, since its not immediately obvious what it does. Consdier even returning a enum instead of Bool, since that's harder to misinterpret

Reply:
I renamed it to `saw_expected_frames`, similar to the `ExpectedFrames` naming